src/Compilers/VisualBasic/Portable/Symbols/ArrayTypeSymbol.vb

```diff
@@ -538,12 +540,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     Return True
                 End Get
             End Property
+
+            Friend Overrides Function WithElementType(elementType As TypeSymbol) As ArrayTypeSymbol
+                Return New SZArray(elementType, CustomModifiers, BaseTypeNoUseSiteDiagnostics, Interfaces)
```

Comment:
Yes. I'll look into updating the interfaces with the new element type.
Seems the same problem exists in C#